File: src/main/java/net/slipp/domain/qna/TemporaryAnswer.java
package net.slipp.domain.qna;

import java.io.Serializable;

public class TemporaryAnswer implements Serializable {
    private static final long serialVersionUID = -4198138126883127272L;
    
    public static final String TEMPORARY_ANSWER_KEY = "temporaryAnswer";
    public static final TemporaryAnswer EMPTY_ANSWER = new EmptyTemporaryAnswer();
    
    private Long questionId;
    private String temporaryAnswer;
    
    public TemporaryAnswer() {
    }

    public TemporaryAnswer(Long questionId, String temporaryAnswer) {
        this.questionId = questionId;
        this.temporaryAnswer = temporaryAnswer;
    }
    
    public Long getQuestionId() {
        return questionId;
    }
    
    public String getTemporaryAnswer() {
        return temporaryAnswer;
    }
    
    public Answer createAnswer() {
        return new Answer(this.temporaryAnswer);
    }
    
    public String generateUrl() {
        return String.format("/questions/%d", this.questionId);
    }
    
    static class EmptyTemporaryAnswer extends TemporaryAnswer {
        private static final long serialVersionUID = 1380330064986704887L;
        
        public Answer createAnswer() {
            return new Answer();
        }
        
        public String generateUrl() {
            return "/";
        }
    }

    @Override
    public String toString() {
        return "TemporaryAnswer [questionId=" + questionId + ", temporaryAnswer=" + temporaryAnswer + "]";
    }
}

File: src/main/scala/net/slipp/domain/qna/TemporaryAnswer.scala
package net.slipp.domain.qna

import java.lang.Long
import java.io.Serializable

import scala.beans.BeanProperty

private class EmptyTemporaryAnswer extends TemporaryAnswer {
  override def createAnswer: Answer = {
    return new Answer
  }

  override def generateUrl: String = {
    return "/"
  }
}

object TemporaryAnswer {
  val TEMPORARY_ANSWER_KEY: String = "temporaryAnswer"
  val EMPTY_ANSWER: TemporaryAnswer = new EmptyTemporaryAnswer
}

class TemporaryAnswer(qId: Long, tAnswer: String) extends Serializable {
  @BeanProperty
  val questionId: Long = qId

  @BeanProperty
  val temporaryAnswer: String = tAnswer

  def this() = this(null, null)

  def createAnswer = new Answer(this.temporaryAnswer)

  def generateUrl: String = {
    return String.format("/questions/%d", this.questionId)
  }

  override def toString: String = {
    return "TemporaryAnswer [questionId=" + questionId + ", temporaryAnswer=" + temporaryAnswer + "]"
  }
}
